lang_GENERIC/analyze/AST_to_IL.ml

```diff
@@ -227,12 +239,9 @@ and pattern_assign_statements env exp eorig pat =
 (* Assign *)
 (*****************************************************************************)
 and assign env lhs _tok rhs_exp eorig =
-  match lhs with
-  | G.Id (_, _) ->
-      let lval = lval env lhs in
-      add_instr env (mk_i (Assign (lval, rhs_exp)) eorig);
-      mk_e (Lvalue lval) lhs
-  | _ -> todo (G.E lhs)
+  let lval = lval env lhs in
```

Comment:
actually lhs can be a tuple, dict, or even object in languages such as JS that supports
destructuring assignment. We should "compile" those in simpler assign.

Reply:
You're right, thanks for pointing it out!